homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,405 @@
+    return value
+
+
+def calc_sum(sensor_values: list[tuple[str, Any]], round_digits: int) -> float:
+    """Calculate a sum of values."""
+    result = 0
+    for _, sensor_value in sensor_values:
+        try:
+            result += sensor_value
+        except TypeError:
+            continue
+
+    value: float = round(result, round_digits)
+    return value
+
+
+class SensorGroup(GroupEntity, SensorEntity):
+    """Representation of a sensor group."""
+
+    _attr_available = False
+    _attr_should_poll = False
+    _attr_icon = "mdi:calculator"
+
+    def __init__(
+        self,
+        unique_id: str | None,
+        name: str,
+        entity_ids: list[str],
+        mode: bool,
+        sensor_type: str,
+        round_digits: int,
+        unit_of_measurement: str | None,
+        state_class: SensorStateClass | None,
+        device_class: SensorDeviceClass | None,
+    ) -> None:
+        """Initialize a sensor group."""
+        self._entity_ids = entity_ids
+        self._sensor_type = sensor_type
+        self._round_digits = round_digits
+        self._attr_state_class = state_class
+        self.calc_state_class: SensorStateClass | None = None
+        self._attr_device_class = device_class
+        self.calc_device_class: SensorDeviceClass | None = None
+        self._attr_native_unit_of_measurement = unit_of_measurement
+        self.calc_unit_of_measurement: str | None = None
+        self._attr_name = name
+        if name == DEFAULT_NAME:
+            self._attr_name = f"{DEFAULT_NAME} {sensor_type}".capitalize()
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+        self._attr_unique_id = unique_id
+        self.mode = all if mode else any
+        self._sensor_attr = SENSOR_TYPE_TO_ATTR[self._sensor_type]
+        self.min_value: float | None = None
+        self.max_value: float | None = None
+        self.mean: float | None = None
+        self.last: float | None = None
+        self.median: float | None = None
+        self.range: float | None = None
+        self.sum: float | None = None
+        self.min_entity_id: str | None = None
+        self.max_entity_id: str | None = None
+        self.last_entity_id: str | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+
+        @callback
+        def async_state_changed_listener(event: Event) -> None:
+            """Handle child updates."""
+            self.async_set_context(event.context)
+            self.async_defer_or_update_ha_state()
+
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, self._entity_ids, async_state_changed_listener
+            )
+        )
+
+        await super().async_added_to_hass()
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the sensor group state."""
+        states = []
+        test_states = []
```

Comment:
Please type these lists:
```py
states: list[State] = []
test_states: list[bool] = []
```

Also, I'd suggest to rename the 2nd list to `valid_states` or `numerical_states`.